### DISCUSSION TITLE
DDECommand [CmdClose] brings window to front

### DISCUSSION THREAD

**Discussion by MarcelSchober** [sumatrapdfreader/sumatrapdf#3106] [General]
Hi,

Is there a way to prevent that the Sumatrapdf window gets in foreground and requests focus, when the [CmdClose] command is executed?

Thanks and Regards
Marcel

**Comment by GitHubRulesOK** [comment]
personally, would not expect it to release focus.
 since after "close" of one tab you need to have it active for next channel command such as open next file
 this was/is a common complaint with acrobat etc.  that it stays active since it makes no sense to close it inefficiently, once resident in memory.
The difference is unlike acrobat, sumatrapdf has a command that can then be applied to the focus "CmdExit"

**Reply by MarcelSchober** [reply]
Thank you very much for your fast reply! 

In my scenario, the sumatrapdf window is in background. When I call [CmdClose] from my app to close the current document in sumatrapdf, the sumatrapdf window becomes active and pops to foreground (which is not usefull in my scenario).

So if sumatrapdf does not have the focus, it should not gain the focus when [CmdClose] is called.

Hope, this makes sense ...

**Reply by GitHubRulesOK** [reply]
it makes sense but why would a user's foreground viewing GUI app be run behind others, It always causes me issues when other apps like explorer are "always on top" as you can't see the working rendered content. so always advise don't run GDI apps as silent it blocks visibility of rendering functions, the point of windows is overlay window panes but generally use side by side.

**Reply by MarcelSchober** [reply]
In my scenario, Sumatrapdf is used as "viewer" on the second screen. My own application is in foreground and when users click in my application, it uses DDE commands to open and close documents in the Sumatrapdf window. But the Sumatrapdf window should not pop to foreground. The focus should stay in my own application.

**Comment by kjk** [comment]
I'm afraid if I change the behavior to not focus, I'll break some valid use case.

I can only suggest a work-around: re-set focus to your own app after sending DDE command.
